projects/fsastorefrontlib/src/cms-components/my-account/quote/quotes/quotes.component.html

```diff
@@ -3,89 +3,179 @@ <h2 class="heading-headline py-3 mt-3 mb-5 text-center">
-                  }}</span>
                 </div>
-                <div class="info-card-data">
-                  <span class="label">{{ 'fscommon.price' | cxTranslate }}</span
-                  >:
-                  <span class="value"
-                    >{{ quote.quotePrice }}
-                    <ng-container *ngIf="quote.paymentFrequency">
-                      / {{ quote.paymentFrequency }}</ng-container
-                    ></span
-                  >
+                <div class="info-card-image">
+                  <span class="image-wrapper text-center">
+                    <cx-media
+                      class="image-item"
+                      [container]="quote.defaultCategory.image"
+                    ></cx-media>
+                  </span>
                 </div>
               </div>
-              <div class="info-card-image">
-                <span class="image-wrapper text-center">
-                  <cx-media
-                    class="image-item"
-                    [container]="quote.defaultCategory.image"
-                  ></cx-media>
-                </span>
-              </div>
-            </div>
-            <ng-container
-              *ngIf="
-                !quote.isExpired &&
-                quote.quoteWorkflowStatus.code !== 'REFERRED'
-              "
-            >
-              <div class="info-card-links">
-                <a
-                  class="link"
-                  [routerLink]="
-                    {
-                      cxRoute: 'quoteDetails',
-                      params: { quoteId: quote.quoteId }
-                    } | cxUrl
-                  "
+              <ng-container
+                *ngIf="
+                  !quote.isExpired &&
+                  quote.quoteWorkflowStatus.code !== 'REFERRED'
+                "
+              >
+                <div
+                  class="info-card-links justify-content-between align-items-center"
                 >
-                  {{ 'fscommon.details' | cxTranslate }}
-                </a>
-                <a class="link" (click)="retrieveQuote(quote)">
-                  {{ 'fscommon.retrieve' | cxTranslate }}
-                </a>
-              </div>
-            </ng-container>
+                  <div
+                    class="checkbox-wrapper"
+                    [class.invisible]="
+                      quotesByCategory[quote.defaultCategory.name]?.length < 2
+                    "
+                  >
+                    <div
+                      class="form-check d-flex align-items-center mb-0"
+                      [class.disabled]="
+                        quoteCodesForCompare?.length > 0 &&
+                        (selectedQuote?.defaultCategory?.code !==
+                          quote.defaultCategory.code ||
+                          (quoteCodesForCompare.length === 2 &&
+                            quoteCodesForCompare.indexOf(quote.cartCode) ===
+                              -1))
```

Comment:
Disables all categories which are different from the selected one and all quotes from the same category if already two are selected.

Reply:
Can we move this to the component method? Something like:
[class.disabled]=disableQuotesFromDifferentCategories(quoteCodesForCompare, selectedQuote, quote)

Reply:
Yes, we can, but calling a method in the template is not good for performance.

Reply:
Maybe move this into a single property in model, and then just reference this property here?

Reply:
Then how to bind that property of boolean type for which quote from the template? This all is happening in for loop in the template. If that property returns true, all checkboxes will be disabled.

Reply:
Then move this logic outside the template into the model, since you are unpacking `quotes$` there as well.

Reply:
So you suggest iterating in the model through quotes even though we do it already through HTML? If I go through for loop in the model and use this expression to decide what should be disabled how to tell a particular quote to be disabled when there is no such attribute on the model?

Reply:
You are already iterating both in the template and in the model:
```
      this.quotes$
        .pipe(
          filter(quotes => quotes.length !== 0),
          tap(quotes => {
```
And that's also something that should be perhaps refactored, if possible, of course. We should avoid having this many loops for the same thing. Right now in the model we are looping twice over the same object (lines 53 and 165), and then again in the template. If we need that logic in the model, maybe then just create a simple array just for quotes inside that loop and use that array in the template? While in that loop perhaps we can check for this other stuff and set some boolean property, which we would then reference in the template for that `disabled` attribute?

Reply:
No, I don't iterate here through quotes, there is no something like `quotes.forEach(quote => { // do something })`, right? And how do you think to reference single boolean property for a specific quote? If that property is true, all quotes will be disabled.